File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.java
class Solution {
    public int solution(int n) {
        int[] tiling = {1, 2, 3};
        
        if (n < 4) {
            return tiling[n-1];
        }
        
        for (int i=4; i<=n; i++) {
            tiling[0] = tiling[1];
            tiling[1] = tiling[2];
            tiling[2] = (int) ((tiling[0] + tiling[1]) % 1_000_000_007L);
        }      
        
        return tiling[2];
    }
}

File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.py
def solution(n):
    tiling = [1, 2, 3]
    
    if n < 4:
        return tiling[n-1]
    
    for i in range(3, n):
        next_tiling = (tiling[-1] + tiling[-2]) % 1_000_000_007
        tiling.append(next_tiling)
    
    return tiling[-1]
